lib/finch.ex

```diff
@@ -224,7 +224,7 @@ defmodule Finch do
 
   """
   @spec stream(Request.t(), name(), acc, stream(acc), keyword) ::
-          {:ok, acc} | {:error, Mint.Types.error()}
+          {:ok, acc} | {:error, Mint.Types.error()} | {:error, Finch.Error.t()}
```

Comment:
Maybe we should use this opportunity to also solve: https://github.com/keathley/finch/issues/54#issuecomment-645381816?

```suggestion
          {:ok, acc} | {:error, Exception.t()}
```

Reply:
Let me know what else would need to be done to address #54.